UMLEditor/UMLEditor/Views/CommandLine/CommandLineController.cs

```diff
@@ -1,513 +1,493 @@
-                                 "\nlist_attributes: List all attributes of a class" +
-                                 "\nlist_relationships: List all relationships of a class\n", ConsoleColor.DarkCyan);
-                break;
-
-            case ("save_diagram"):
-                if (arguments.Count == 1)
-                {
-                    string fileString = arguments.ElementAt(0);
-                    if (!fileString.EndsWith(".json"))
-                    {
-                        PrintColoredLine("Please use .json file extension", ERROR_COLOR);
-                        break;
-                    }
-
-                    _activeFile!.SaveDiagram(ref _activeDiagram!, arguments[0]);
-
-
-                    PrintColoredLine($"Current diagram saved to '{arguments[0]}'", SUCCESS_COLOR);
-                }
-
-                else
-                {
-                    PrintColoredLine("save_diagram [fileName]", ERROR_COLOR);
-                }
-
-                break;
-
-            case ("load_diagram"):
-                if (arguments.Count == 1)
-                {
-                    string fileString = arguments[0];
-                    if (!fileString.EndsWith(".json"))
-                    {
-                        PrintColoredLine("Please provide a file with a .json file extension", ERROR_COLOR);
-                        break;
-                    }
-
-                    _activeDiagram = _activeFile!.LoadDiagram(arguments[0]);
-
-
-                    PrintColoredLine($"Diagram '{arguments[0]}' loaded.", SUCCESS_COLOR);
-                }
-
-                else
-                {
-                    PrintColoredLine("load_diagram [fileName]", ERROR_COLOR);
-                }
-
-                break;
-
-            case ("exit"):
-                Environment.Exit(0);
-                break;
-
-            default:
-                PrintColoredLine("Please provide a valid command", ERROR_COLOR);
-                break;
-        }
-    }
-
-    /// <summary>
-    /// Prints the provided text in a specified color.
-    /// </summary>
-    /// <param name="text">The text to print.</param>
-    /// <param name="inColor">The color you want the text printed in.</param>
-    private static void PrintColoredLine(string text, ConsoleColor inColor = ConsoleColor.White)
-    {
-        ConsoleColor oldTextColor = Console.ForegroundColor;
-
-        Console.ForegroundColor = inColor;
-        Console.WriteLine(text);
-
-        Console.ForegroundColor = oldTextColor;
-    }
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using UMLEditor.Classes;
+using UMLEditor.Interfaces;
+#pragma warning disable CS0162
+
+namespace UMLEditor.Views.CommandLine;
+
+/// <summary>
+/// Controller for the Command Line View
+/// </summary>
+public class CommandLineController
+{
+
+    
+    private Diagram? _activeDiagram;
+    private IDiagramFile? _activeFile;
```

Comment:
The activeDiagram and activeFile don't need to be nullable. You initialize them in the constructor.